services/common/serv_helper.js

```diff
@@ -52,10 +52,10 @@
         var dest = cloned[key];
         var source = obj[key];
         for (var i = 0, l = source.length; i < l; i++) {
-          dest[i] = cloneObject(source[i]);
+          dest[i] = cloneObject(source[i], recursive);
         }
       } else if (typeof obj[key] === 'object') {
-        cloned[key] = !recursive && cloneObject(obj[key]) || obj[key];
+        cloned[key] = recursive && cloneObject(obj[key], recursive) || obj[key];
```

Comment:
Now the function does what is expected to do. Notice that the recursive parameter is used to clone the object recursively so the previous line `!recursive` doesn't have sense, at least for me. Let's assume that the flag is true when we need to clone the object recursively otherwise the object will just be copied.